sphinx_immaterial/cpp_domain_fixes.py

```diff
@@ -751,7 +751,7 @@ def cross_link_single_parameter(
                 "Parameter name %r does not match any of the parameters "
                 "defined in the signature: %r",
                 param_name,
-                list(all_params.keys()),
+                obj_content.parent.children[0].astext(),
```

Comment:
There can be more than one signature, so it would be better to include all of them.

The full list of signatures can be obtained from `obj_content.parent.children[:-1]`  --- the last child is always `obj_content`.

Reply:
I did that and it returned the documentation nodes for the node as well. Also, `astext()` is specific to a node, not a list of nodes.

Reply:
You are saying that even with excluding the last child, of `obj_content.parent`, you still got the documentation?

In the case that more than one signature is specified, the valid parameter name set is the union of parameter names from every signature, so just displaying the first signature may be confusing.

I would suggest changing to:

```python
            logger.warning(
                "Parameter name %r does not match any of the parameters %r "
                "defined in the signatures %r",
                param_name,
                list(all_params.keys()),
                [signode.astext() for signode in signodes],
                location=param_node,
            )
```

For simplicity, you could just add `signodes` as an additional parameter to `_add_parameter_documentation_ids`.

Reply:
using
```py
[n.astext() for n in obj_content.parent.children[-1:]]
```
The warning we've been getting (about the optional syntax) looks like
```
sphinx-immaterial\docs\additional_samples.rst:234: WARNING: Parameter name 'limit' does not match any of the parameters defined in the signature: ['Format the exception with a traceback.\n\nParameters\n\netype\n\nexception type\n\nvalue\n\nexception value\n\ntb\n\ntraceback object\n\nlimit : integer or None\n\nmaximum number of stack frames to show\n\nReturn type\n\nlist of strings']
```

Reply:
`obj_content.parent.children[-1:]` means the list containing only the last child --- we want `obj_content.parent.children[:-1]` which means the list containing all but the last child.

Reply:
opps. I didn't realize I put the colon in the wrong place. Uding
```py
            logger.warning(
                "Parameter name %r does not match any of the parameters "
                "defined in the signature(s): %r",
                param_name,
                [n.astext() for n in obj_content.parent.children[:-1]],
                location=param_node,
            )
```
produced the desired output:
```
sphinx-immaterial\docs\additional_samples.rst:234: WARNING: Parameter name 'limit' does not match any of the parameters defined in the signature: ['format_exception(etype, value, tb, [limit=None])']
```

I'm going to test this with the C++ domain and commit if the results are consistent.

Reply:
resolved

Reply:
I still think it would be helpful to include the list of allowed parameters as well --- for example, in the above case, it may not be obvious that we cannot properly parse `limit`.

Reply:
With these requested changes, the parameters are listed as provided to the signature. The whole reason for changing this is because the C/C++ domain can have an unhelpful empty list of parameters.

Reply:
Ok, I've added a list of "known parameters" to the warnings, which uses the original approach.

Reply:
Thanks